### PR TITLE
Refactor Serenity standard framework after hook

### PR DESCRIPTION
**Pull request by Shorakie**
This PR, refactors the after hook function to be more readable

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -300,64 +300,73 @@ fn display_dispatch_error(err: DispatchError) -> String {
...
-                        &err
-                    );
+    let err = if let Err(err) = result { err } else { return };
+    // handle UserErr
+    if let Some(err) = err.downcast_ref::<UserErr>() {
+        match err {
+            UserErr::MentionedUserNotFound => {
+                msg.reply_error(&ctx, "No user found with that name")
+                    .await
+                    .ok();
```
Comment (druskus20) [inline src/main.rs:310]: This is really not better than `let _ =` imo. No need to turn the result into an Option. Less expressive.

File: src/main.rs [hunk 2]
```diff
@@ -300,64 +300,73 @@ fn display_dispatch_error(err: DispatchError) -> String {
...
+            }
+            UserErr::Other(issue) => {
+                msg.reply_error(&ctx, format!("Error: {}", issue))
+                    .await
+                    .ok();
+            }
+        }
+        return;
```
Comment (druskus20) [inline src/main.rs:323]: I don't like this. Early returns are really not needed, especially when we are just pattern matching multiple times, I don't think it is better than indentation. Technically what one could do is abstract chunks into separate functions while keeping the overall structure the same. However, if we're not going to reuse those functions there's really no point in doing so.

File: src/main.rs [hunk 3]
```diff
@@ -300,64 +300,73 @@ fn display_dispatch_error(err: DispatchError) -> String {
...
+                error.message = %err,
+                "Internal error [handling {}]: {} ({:#?})",
+                command_name,
+                &err,
+                &err
+            );
+            return;
+        }
+    };
+    match serenity_err {
```
Comment (druskus20) [inline src/main.rs:352]: Reiterating what I've said before, I really don't like this `match` here.

**Review by druskus20 — COMMENTED** [review]
I sadly don't think this improves much, left a couple of comments explaining why.

**Comment by druskus20** [comment]
I.. really dont know how this is better than the original though... we went from using `match` everywhere to mix-matching (no pun intended) both `match` and `if let`... 
Otherwise, I would probably like to read an explanation of where the improvement is supposed to be.
But maybe @elkowar has a different opinion

**Review by druskus20 — COMMENTED** [review]
I.. really dont know how this is better than the original though... we went from using `match` everywhere to mix-matching (no pun intended) both `match` and `if let`... 
Otherwise, I would probably like to read an explanation of where the improvement is supposed to be.
But maybe @elkowar has a different opinion

**Comment by elkowar** [comment]
> I.. really dont know how this is better than the original though... we went from using `match` everywhere to mix-matching (no pun intended) both `match` and `if let`... 
> Otherwise, I would probably like to read an explanation of where the improvement is supposed to be.
> But maybe @elkowar has a different opinion 
> 

I agree with you on this, I don't really think the if let is an improvement here :/

**Comment by Shorakie** [comment]
I understand, ill close the PR.

**Pull request closed without merging**